sound/soc/sof/ipc.c

```diff
@@ -241,10 +243,64 @@ static int tx_wait_done(struct snd_sof_ipc *ipc, struct snd_sof_ipc_msg *msg,
 	return ret;
 }
 
-/* send IPC message from host to DSP */
 int sof_ipc_tx_message(struct snd_sof_ipc *ipc, u32 header,
-		       void *msg_data, size_t msg_bytes, void *reply_data,
-		       size_t reply_bytes)
+			void *msg_data, size_t msg_bytes, void *reply_data,
+			size_t reply_bytes)
+{
+	struct sof_ipc_large_hdr *hdr;
+	uint32_t num_msg = 0;
+	uint32_t offset = 0;
+	uint32_t send_bytes = 0;
+	int ret = -1;
+	int size;
+	int i;
+
+	/* if bigger than max size, chop to smaller pieces */
+	if (msg_bytes > SOF_IPC_MSG_MAX_SIZE) {
```

Comment:
You can win an indentation level if you check for msg < SOF_IPC_MSG_MAX_SIZE and return directly here.